docs/user/azure/install_upi.md

```diff
@@ -0,0 +1,469 @@
+
+```sh
+export VHD_URL=`curl -s https://raw.githubusercontent.com/openshift/installer/master/data/data/rhcos.json | jq -r .azure.url`
+```
+
+Create a blob storage container and copy the image:
+
+```sh
+az storage container create --name vhd --account-name sa${CLUSTER_NAME}
+az storage blob copy start --account-name sa${CLUSTER_NAME} --account-key $ACCOUNT_KEY --destination-blob "rhcos.vhd" --destination-container vhd --source-uri "$VHD_URL"
+```
+
+To track the progress, you can use:
+
+```sh
+status="unknown"
+while [ "$status" != "success" ]
+do
+    status=`az storage blob show --container-name vhd --name "rhcos.vhd" --account-name sa${CLUSTER_NAME} --account-key $ACCOUNT_KEY -o tsv --query properties.copy.status`
+done
+```
+
+Store the URL of the copied image for later use:
+
+```sh
+export VHD_BLOB_URL=`az storage blob url --account-name sa${CLUSTER_NAME} --account-key $ACCOUNT_KEY -c vhd -n "rhcos.vhd" -o tsv`
+```
+
+### Upload the ignition file
+
+Create a blob storage container and upload the bootstrap.ign file:
+
+```sh
+az storage container create --name files --account-name sa${CLUSTER_NAME} --public-access blob
+az storage blob upload --account-name sa${CLUSTER_NAME} --account-key $ACCOUNT_KEY -c "files" -f "bootstrap.ign" -n "bootstrap.ign"
+export BOOTSTRAP_URL=`az storage blob url --account-name sa${CLUSTER_NAME} --account-key $ACCOUNT_KEY -c "files" -n "bootstrap.ign" -o tsv`
+```
+
+## Deployment
+
+The key part of this UPI deployment are the [Azure Resource Manager][azuretemplates] templates, which are responsible
+for deploying most resources. They're provided as a few json files named following the "NN_name.json" pattern. In the
+next steps we're going to deploy each one of them in order, providing the expected parameters.
+
+### Deploy the VPC
+
+```sh
+az group deployment create -g $RESOURCE_GROUP \
+  --name 01_${CLUSTER_NAME} \
+  --template-file "01_vpc.json"
+```
+
+### Deploy the storage account
+
+```sh
+az group deployment create -g $RESOURCE_GROUP \
+  --name 02_${CLUSTER_NAME} \
+  --template-file "02_storage.json" \
+  --parameters vhdBlobURL="${VHD_BLOB_URL}"
+```
+
+### Deploy the load balancers
+
+Create the public IP addresses:
+
+```sh
+az network public-ip create -g $RESOURCE_GROUP -n $CLUSTER_NAME --allocation-method static --sku Standard
+az network public-ip create -g $RESOURCE_GROUP -n ${CLUSTER_NAME}app --allocation-method static --sku Standard
+```
+
+Deploy them:
+
+```sh
+az group deployment create -g $RESOURCE_GROUP \
+  --name 03_${CLUSTER_NAME} \
+  --template-file "03_infra.json"
+```
+
+Create DNS records for the public load balancer:
+
+```sh
+export PUBLIC_IP=`az network public-ip list -g $RESOURCE_GROUP --query "[?name=='${CLUSTER_NAME}'] | [0].ipAddress" -o tsv`
+export PUBLIC_IP_APPS=`az network public-ip list -g $RESOURCE_GROUP --query "[?name=='${CLUSTER_NAME}app'] | [0].ipAddress" -o tsv`
+
+az network dns zone create -g $RESOURCE_GROUP -n ${CLUSTER_NAME}.${BASE_DOMAIN}
```

Comment:
Per my experience in IPI testing, public base domain zone maybe already exists in other resource group, such in QE's azure subscription, the public base domain zone is `qe.azure.devcluster.openshift.com`, which is existing in `os4-common` resource group. 
So I do not think here create a new domain zone in `cluster` resource group is a necessary step, user only need create A record for public api/apps url in `os4-common` resource group.
I think upi should be keep consistent like ipi install.

Reply:
That will not always be the case in the customer's environment, but I agree that scenario must be covered. Docs updated.